libs/components/popovers/testing/src/dropdown/harness/dropdown-harness.ts

```diff
@@ -0,0 +1,125 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+
+import { SkyDropdownHarnessFilters } from './dropdown-harness.filters';
+import { SkyDropdownMenuHarness } from './dropdown-menu-harness';
+
+export class SkyDropdownHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-dropdown';
+
+  #documentRootLocator = this.documentRootLocatorFactory();
+
+  #getDropdownButton = this.locatorFor('.sky-dropdown-button');
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyDropdownHarness` that meets certain criteria.
+   */
+  public static with(
+    filters: SkyDropdownHarnessFilters
+  ): HarnessPredicate<SkyDropdownHarness> {
+    return SkyDropdownHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Clicks the dropdown button.
+   */
+  public async clickDropdownButton(): Promise<void> {
+    (await this.#getDropdownButton()).click();
+  }
+
+  /**
+   * Gets the aria-label value.
+   */
+  public async getAriaLabel(): Promise<string | null> {
+    return (await this.#getDropdownButton()).getAttribute('aria-label');
+  }
+
+  /**
+   * Gets the dropdown button style.
+   */
+  public async getButtonStyle(): Promise<string> {
+    const button = await this.#getDropdownButton();
+
+    if (await button.hasClass('sky-btn-primary')) {
+      return 'primary';
+    }
+
+    if (await button.hasClass('sky-btn-link')) {
+      return 'link';
+    }
+
+    return 'default';
+  }
+
+  /**
+   * Gets the dropdown button type.
+   */
+  public async getButtonType(): Promise<string> {
+    const button = await this.#getDropdownButton();
+
+    if (await button.hasClass('sky-dropdown-button-type-context-menu')) {
+      return 'context-menu';
+    }
+
+    return 'select';
+  }
+
+  /**
+   * Gets the dropdown menu component.
+   */
+  public async getDropdownMenu(): Promise<SkyDropdownMenuHarness | null> {
```

Comment:
This should never return `null`. If the dropdown has a menu ID, but the menu cannot be found, it should throw an error (similar to what you're doing with the `getItems` method on the menu harness).
```
public async getDropdownMen(): Promise<SkyDropdownMenuHarness>
```
This will also allow the consumer to not have to do a null check when accessing the menu. For example, instead of:

```
await (await dropdownHarness.getDropdownMenu())?.clickOut();
```
consumers can now do this:

```
await (await dropdownHarness.getDropdownMenu()).clickOut();
```